src/Cassandra.IntegrationTests/Core/TypeSerializersTests.cs

```diff
@@ -298,5 +302,539 @@ private byte[] serializeForDynamicType(params object[] vals)
+                Assert.AreEqual(1, elem.Count);
+                var subElem = elem[0].ToList();
+                Assert.AreEqual(3, subElem.Count);
+                for (var idxj = 0; idxj < vectorList[idx].Count; idxj++)
+                {
+                    assertFn(vectorList[idx][idxj], subElem[idxj]);
+                }
+            }
+        }
+
+        private void SetupVectorUdtSchema()
+        {
+            if (Interlocked.CompareExchange(ref vectorSchemaSetUp, 1, 0) == 0)
+            {
+                Session.Execute("create type if not exists fixed_type (a int, b int)");
+                Session.Execute("create type if not exists mixed_type_one (a int, b varint)");
+                Session.Execute("create type if not exists mixed_type_two (a varint, b int)");
+                Session.Execute("create type if not exists var_type (a varint, b varint)");
+                Session.Execute("create type if not exists complex_vector_udt (a vector<int, 3>, b vector<varint, 3>)");
+            }
+            Session.UserDefinedTypes.Define(
+                UdtMap.For<FixedType>("fixed_type"),
+                UdtMap.For<MixedTypeOne>("mixed_type_one"),
+                UdtMap.For<MixedTypeTwo>("mixed_type_two"),
+                UdtMap.For<VarType>("var_type"),
+                UdtMap.For<ComplexVectorUdt>("complex_vector_udt"));
+        }
+
+        public class FixedType
+        {
+            protected bool Equals(FixedType other)
+            {
+                return a == other.a && b == other.b;
+            }
+
+            public override bool Equals(object obj)
+            {
+                if (obj is null) return false;
+                if (ReferenceEquals(this, obj)) return true;
+                if (obj.GetType() != GetType()) return false;
+                return Equals((FixedType)obj);
+            }
+
+            public override int GetHashCode()
+            {
+                unchecked
+                {
+                    return (a * 397) ^ b;
+                }
+            }
+
+            public int a { get; set; }
+
+            public int b { get; set; }
+        }
+
+        public class MixedTypeOne
+        {
+            protected bool Equals(MixedTypeOne other)
+            {
+                return a == other.a && b.Equals(other.b);
+            }
+
+            public override bool Equals(object obj)
+            {
+                if (obj is null) return false;
+                if (ReferenceEquals(this, obj)) return true;
+                if (obj.GetType() != GetType()) return false;
+                return Equals((MixedTypeOne)obj);
+            }
+
+            public override int GetHashCode()
+            {
+                unchecked
+                {
+                    return (a * 397) ^ b.GetHashCode();
+                }
+            }
+
+            public int a { get; set; }
+
+            public BigInteger b { get; set; }
+        }
+
+        public class MixedTypeTwo
```

Comment:
I'm going to be honest, I looked at Bret's python driver PR for some of these test cases and these types actually come from there so I didn't spend much time thinking about the test cases that involve these types and their names. 

Looking a bit more into this, it does appear that having all these types is a bit redundant since UDT as a vector subtype is always handled the same way regardless of the types in that UDT (and the names might be a bit misleading as you noted).

@absurdfarce do you have any thoughts on this? I'm leaning towards changing these UDT names but keeping them since the tests are already functional and removing tests feels awkward

Reply:
My intent with the Python test was to make it as robust as possible in the hopes of catching future regression.  I agree with @SiyaoIsHiding that UDTs will always be encoded as variable size type so these distinctions shouldn't matter for the current implementation.  It's more of a hedge against future changes.